Docs build cleanup

bmispelon opened the pull request:
Result of my Pycon CZ sprint with @HonzaKral.

This adds some tests and simplifies some of the logic in `update_docs`.

One practical result is that the elasticsearch indexing now needs to be done in a separate step by running the `update_index --delete` command.

This should fix #534.

timgraham commented on requirements/common.txt:
```diff
@@ -5,7 +5,8 @@ django-flat-theme==0.9.3
 django-registration-redux==1.2
-elasticsearch-dsl==0.0.4
+elasticsearch-dsl==0.0.9
+elasticsearch>=1.0.0,<2.0.0
```
Why is this needed?

HonzaKral replied:
Elasticsearch (the server) 2.0 is slightly backwards incomaptible with 1.x currently used by the site. To mirror that there are also two versions of the python client - 1.x and 2.x. We just need to fix the correct one in place to correspond with the ES version we are using. Otherwise it would default to newest one.

timgraham commented:
With the current deployment scripts, we don't have an easy way to run one management command right after another completes (if this is important). Maybe creating a new command that calls "update_docs --verbosity=0" (the current command) and "update_index --delete" is needed?

bmispelon commented:
Are the deployment scripts available somewhere pulbic?

bmispelon commented:
I pushed another commit with a new management command that combines the two. Would that work?

bmispelon commented:
Turns out the new `update_docs` wasn't actually working (because of two issues with using `Path` objects instead of strings).

I fixed them, rebased and pushed (also tested the command locally).

timgraham commented:
In my local testing:
```
$ python manage.py update_docs_and_index
...
  File "/home/tim/code/djangoproject.com/docs/management/commands/update_index.py", line 32, in handle
    self.log('Successfully indexed item %s' % id_)
  File "/home/tim/code/djangoproject.com/docs/management/commands/update_index.py", line 23, in log
    if self.verbosity >= level:
TypeError: unorderable types: int() >= str()
```
Just in case it matters, I noticed I have `elasticsearch 1.7.0` instead of the latest `1.X` which looks like `elasticsearch 1.9.0`.

timgraham commented on docs/management/commands/update_docs_and_index.py:
```diff
@@ -0,0 +1,11 @@
+
+class Command(BaseCommand):
+    """
+    Update the docs then re-index them in elasticsearch.
```
reindex

timgraham commented on docs/models.py:
```diff
@@ -117,6 +119,36 @@ def scm_url(self):
+        """
+        Sync the given list of documents (decoded fjson files from sphinx) to
+        the database. Deletes all the release's documents first then
+        re-inserts them as needed.
```
reinserts

bmispelon commented:
The `update_index` command was actually mis-handling the `verbosity` argument.

That should be fixed now.
